src/libhipSYCL/application.cpp

```diff
@@ -25,54 +25,15 @@
  * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
-#include <iostream>
-#include <cassert>
+#include "CL/sycl/detail/application.hpp"
 
-#include <CL/sycl.hpp>
+namespace cl {
+namespace sycl {
+namespace detail {
 
+std::unique_ptr<runtime> application::rt = std::make_unique<runtime>();
```

Comment:
While debugging issue #47, a question popped up that you perhaps already thought about when you wrote this code and already have an answer to. I was wondering if it could happen that actually we end up with two `runtime` objects: One inside `libhipSYCL` and one inside the user application by means of the included `application.hpp`. In practice this doesn't happen on Linux because the runtime object is exported from `libhipSYCL`. My question is: Is this portable and also works on other platforms? Would it be beneficial to also move the other functions inside `application` to the `cpp` file to make this more robust?

Reply:
You're right, this might be an issue on Windows (or which *other platforms* are we talking about?). I actually thought about moving the member's implementations to the new source file, but apparently then forgot about it :[. 

Are you planning on adding Windows support?

Reply:
Yes, I'm mostly talking about Windows. While I'm not personally interested in windows support, I also wouldn't like to intentionally break it (especially with such subtle issues) since, at least in principle hipSYCL should be able to run on Windows (the bigger problem would likely be clang-cuda and ROCm).
I think I would prefer if we could just move the member functions to the cpp to make to make this more robust against such subtle problems in case someone decides at some point to try using it on Windows :)

Reply:
I just made a PR with these changes.

Also, regarding clang-cuda and Windows: While investigating the Clang bug I came across [this commit](https://github.com/llvm/llvm-project/commit/66a8b07cd9) - it looks like work is being done to support Windows in Clang 9 :)

Reply:
Nice, that's good to know :)